demo/KevinDemo/MainView.swift

```diff
@@ -10,56 +10,210 @@ import Localize_Swift
+                            openLink: { linkString in
+                                viewModel.openLink(linkString)
+                            }
+                        )
+                        
+                        Button(action: {
+                            viewModel.onDonateButtonTapped()
+                        }) {
+                            Text(String(
+                                format: "donate_button_title".localized(),
+                                viewModel.viewState.amountString,
+                                "currency_eur".localized())
+                            ).style(.buttonTitle)
+                        }
+                        .frame(
+                            minWidth: 200,
+                            maxWidth: .infinity,
+                            minHeight: 48.0,
+                            maxHeight: 48.0,
+                            alignment: .center
+                        )
+                        .background(viewModel.viewState.isDonateButtonDisabled ? Color.init("DisabledButton") : Color.init("AccentColor"))
+                        .cornerRadius(10)
+                        .disabled(viewModel.viewState.isDonateButtonDisabled)
+                    }
+                    .frame(
+                        minWidth: 0,
+                        maxWidth: .infinity,
+                        minHeight: 0,
+                        maxHeight: .infinity,
+                        alignment: .topLeading
+                    )
+                    .padding(16)
                 }
-            })
-            
-            .alert(isPresented: $viewModel.viewState.showMessage, content: {
-                Alert(
-                    title: Text(viewModel.viewState.messageTitle!),
-                    message: Text(viewModel.viewState.messageDescription!),
-                    dismissButton: .default(Text("action_ok".localized()))
+            }
+        }
+        .halfASheet(
+            isPresented: $viewModel.viewState.isCountrySelectorPresented,
+            content: {
+                KevinCountrySelector(
+                    countyCodes: viewModel.viewState.countryCodes,
+                    onCountrySelected: { selectedCountryCode in
+                        viewModel.onCounrtyCodeSelected(selectedCountryCode)
+                    }
                 )
-            })
-            
-            ProgressView().padding(30).opacity(viewModel.viewState.isLoading ? 1 : 0)
-            Spacer()
+            },
+            configuration: HalfASheetConfiguration(
+                appearanceAnimationDuration: 0.2,
+                backgroundColor: .white,
+                height: .proportional(0.7),
+                contentInsets: EdgeInsets(top: 0, leading: 0, bottom: 0, trailing: 0),
+                allowsDraggingToDismiss: false,
+                allowsButtonDismiss: false
+            )
+        )
+        .onTapGesture {
+            viewModel.endTextInput()
```

Comment:
this is more of a presentation logic. How about making a view extension like it is presented [here](https://www.hackingwithswift.com/quick-start/swiftui/how-to-dismiss-the-keyboard-for-a-textfield) and hiding keyboard from here, but not in vm?